rtlive/tests.py

```diff
@@ -88,6 +88,69 @@ def test_unsupported_country(self):
             )
 
 
+class TestDataDE:
+    def test_estimate_test_percentages(self):
+        from rtlive.sources import data_de
+
+        df = pandas.DataFrame(
+            index=pandas.MultiIndex.from_product([
+                ["RegionA", "RegionB", "RegionC", "all"],
+                pandas.date_range("2020-11-01", periods=14),
+            ], names=["region", "date"]),
+            data={
+                "new_tests": numpy.nan
+            }
+        )
+        na = numpy.nan
+        df.loc["RegionA", "new_tests"] = [10, 10, 10, 10, 10, 20, 20, 20, na, na, na, na, na, na]
+        df.loc["RegionB", "new_tests"] = [10, 10, 10, 10, 30, 20, 20, 30, 30, 30, na, 12, na, na]
+        df.loc["RegionC", "new_tests"] = [40, 40, 40, 10, 30, 20, 0., 17, 40, na, 20, na, na, na]
+        df.loc["all", "new_tests"] =     [60, 60, 60, 30, 70, 60, 40, 67, 70, 30, 20, 12, na, na]
+        df.loc["all", "new_tests"] = df.loc["all", "new_tests"].values * 1.3 # plus a 30 % dark figure
+        # the last 7 days with data for all:  |=========================|
+        # total tests in that window: 100 + 130 + 157 = 387
```

Comment:
Looking at this test I noticed, that the function makes the assumption, that if there is a day for which all regions have known data that all data is known up until that point.

That might not be true, e.g. the data could look like this:
```
        df.loc["RegionA", "new_tests"] = [10, 10, 10, na, na, na, na, 20, na, na, na, na, na, na]
        df.loc["RegionB", "new_tests"] = [10, 10, na, na, na, na, na, 30, 30, 30, na, 12, na, na]
        df.loc["RegionC", "new_tests"] = [40, 40, 40, na, na, na, na, 17, 40, na, 20, na, na, na]
```

Probably not worth fixing that, because the function is unused anyway and i assume the likelihood of this problem appearing in the data is very small.

Reply:
Excellent point I'm adding comments about that.